pkg/config/client/transport.go

```diff
@@ -0,0 +1,38 @@
+package client
+
+import (
+	"net/http"
+
+	"k8s.io/client-go/rest"
+)
+
+// AnonymousClientConfigWithWrapTransport returns a copy of the given config with all user credentials (cert/key, bearer token, and username/password) and custom transports (Transport) removed.
+// This function preserves WrapTransport for clients that care about custom HTTP behavior.
+func AnonymousClientConfigWithWrapTransport(config *rest.Config) *rest.Config {
+	newConfig := rest.AnonymousClientConfig(config)
+	newConfig.WrapTransport = config.WrapTransport
```

Comment:
I literally asked for that upstream :wink:

Reply:
heh :)